scripts/lib/CIME/XML/env_mach_specific.py

```diff
@@ -298,6 +311,8 @@ def _load_modules_generic(self, modules_to_load):
         for key in newenv:
             if key in os.environ and key not in newenv:
                 del(os.environ[key])
+            elif key in os.environ and os.environ[key] == newenv[key]:
```

Comment:
@vanroekel , I believe this should fix your issue. It won't bother trying to set LS_COLORS. Please confirm.

Reply:
@jgfouca I still have the same error with this line included.  It looks like the regular expression match above splits out the value after the last equal in the string, this gives keys in newenv with '=' in them.  A solution that works for me is commented above.

Reply:
git won't let me comment above.  so I'll do so here.  If I change line 293 above from 
```
m=re.match(r'^(\S+)=(\S+)\s*;*\s*$',line)
```
to
```
m=re.match(r'^(\S+?)=(\S+)\s*;*\s*$',line)
```
This will stop at the first equal instead of the final one and my problem is fixed.